### DISCUSSION TITLE
How can I use the code to execute the "format" command

### DISCUSSION THREAD

**Discussion by NightMingGit** [TypeFox/monaco-languageclient#516] [Q&A]
How can I use the code to execute the "format" command

**Comment by kaisalmen** [comment]
@NightMingGit do you mind sharing your code? Do you have a client that runs in the browser and successfully connects to the Java LSP?

Do you see the "Format Document" when you right-click in the editor? 

Programmatically you can fire commands as such:
```typescript
vscode.commands.executeCommand('myCommand', // pass fitting arguments required for specific commands )
.then(() => {
    // do what you need after invocation
});
```

**Comment by kaisalmen** [comment]
Why did you open up a new issue when this question could just be added to the existing discussion?

**Comment by pswai** [comment]
> How can I use the code to execute the "format" command

Not sure if you have figured it out. For anyone landed on this thread, this is what i did:

By right, the following should work. However the gotcha is that the editor has to be in focus. So this won't work say if you have an external button to trigger formatting.

```ts
const documentUri = vscode.window.activeTextEditor.document.uri;
vscode.commands.executeCommand("editor.action.formatDocument", documentUri);
```

For that scenario to work, you need to include the '@codingame/monaco-vscode-editor-service-override' service.

```ts
import { useOpenEditorStub } from 'monaco-editor-wrapper/vscode/services';
import getEditorServiceOverride from '@codingame/monaco-vscode-editor-service-override';

const defaultUserConfig: UserConfig = {
  wrapperConfig: {
    serviceConfig: {
      userServices: {
        ...getEditorServiceOverride(useOpenEditorStub),
      },
    },
    // ...
  },
};
```

I am not sure what it does internally, but this is what I read in the doc
<img width="848" alt="image" src="https://github.com/user-attachments/assets/f46a1361-9db4-48db-af2a-8231f31e30e3">

Also the Langium example did this. Hope this helps!
